src/errors.rs

```diff
@@ -0,0 +1,32 @@
+use core::fmt;
+
+#[derive(Debug)]
```

Comment:
@JelteF 

> Open question 1: Do we to derive `Debug`?

Yes. We do want to derive `Debug` (and `Clone` too), because it's so unpleasant situation when third-party lib doesn't provide a `Debug` impl for a type, and you should squat-around it with wrappers in case you just want to put `dbg!()`, and I hit it so many times 🤬

> Maybe not, because it requires all types for which TryFrom is derived to also have a Debug implementation. Can we make this conditionally derive Debug only when T implements Debug itself?

Yup. Event standard `#[derive(Debug)]` will generate a proper `impl<T: Debug> Debug for MyError<T>` in this situation. We don't need to do something additional here.

> Open question 2: Do we want to implement the Error trait for this error type? This doesn't work for no_std currently, but we could make std an optional default enabled feature.

Yes. Definitely.

Reply:
Added deriving `Clone` and `Copy` in https://github.com/JelteF/derive_more/pull/220/commits/a6e68dfb07552ba881d658cd2d23991b1eaf5ce5. These are conditional in the same way as `#[derive(Debug)]`.

Reply:
> Yup. Event standard #[derive(Debug)] will generate a proper impl<T: Debug> Debug for MyError<T> in this situation. We don't need to do something additional here.

Yes of course, it seems I was misreading the following error message when removing `Debug` from `MixedData`:
```
error[E0277]: `MixedData` doesn't implement `Debug`
    --> src/lib.rs:250:42
     |
14   | assert_eq!(123u32, mixed_int1.try_into().unwrap());
     |                                          ^^^^^^ `MixedData` cannot be formatted using `{:?}`
     |
     = help: the trait `Debug` is not implemented for `MixedData`
     = note: add `#[derive(Debug)]` to `MixedData` or manually `impl Debug for MixedData`
     = help: the trait `Debug` is implemented for `TryIntoError<T>`
     = note: required because of the requirements on the impl of `Debug` for `TryIntoError<MixedData>`
 ```

the reason it's complaining here is because `unwrap` requires an `E` that implements `Debug`